command/server.go

```diff
@@ -446,9 +446,35 @@ func (c *ServerCommand) parseConfig() (*server.Config, []configutil.ConfigError,
 		c.UI.Warn("WARNING: Entropy Augmentation is not supported in FIPS 140-2 Inside mode; disabling from server configuration!\n")
 		config.Entropy = nil
 	}
-
 	entCheckRequestLimiter(c, config)
 
+	if config != nil {
```

Comment:
I'll be the first to admin that this code smells. I considered checking for `disable_mlock` in each individual config file, but the value only has to exist in one config and since they are parsed individually, that didn't pan out. This is the only place it made sense to me, but I am open to other ideas.

The additional check for `config == nil` also makes me a bit uncomfortable. Because it's also used for checking Entropy Augmentation, we may need to look at this apart from this PR.